ext/rbs_extension/parser.c

```diff
@@ -1092,10 +1092,13 @@ static VALUE parse_intersection(parserstate *state) {
   rg.start = state->next_token.range.start;
 
   VALUE type = parse_optional(state);
-  VALUE intersection_types = rb_ary_new();
+  VALUE intersection_types = EMPTY_ARRAY;
 
-  rb_ary_push(intersection_types, type);
   while (state->next_token.type == pAMP) {
+    if (intersection_types == EMPTY_ARRAY) {
```

Comment:
We can early return before allocating `intersection_types`, and I think it's better code organization.

```c
...

VALUE type = parse_optional(state);

if (state->next_token.type != pAMP) {
  return type;
}

VALUE intersection_types = rb_ary_new();
rb_ary_push(intersection_types, type);

...
```